tracing/plugin/otlp_test.go

```diff
@@ -0,0 +1,80 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"testing"
+)
+
+// TestNewExporter runs tests with different combinations of configuration for NewExporter function
+func TestNewExporter(t *testing.T) {
+
+	for _, testcase := range []struct {
+		input  OTLPConfig
+		output error
+	}{
+		//Test http/protobuf protocol, expect no error
+		{
+			input: OTLPConfig{Endpoint: "http://localhost:4318",
+				Protocol: "http/protobuf",
+				Insecure: false},
+			output: nil,
+		},
+		//Test default protocol, expect no error
+		{
+			input: OTLPConfig{Endpoint: "http://localhost:4318",
+				Protocol: "",
+				Insecure: false},
+			output: nil,
+		},
+		//Test grpc protocol, expect no error
+		{
+			input: OTLPConfig{Endpoint: "http://localhost:4317",
+				Protocol: "grpc",
+				Insecure: false},
+			output: nil,
+		},
+		//Test http/json protocol which is not supported, expect not supported error
+		{
+			input: OTLPConfig{Endpoint: "http://localhost:4318",
+				Protocol: "http/json",
+				Insecure: false},
+			output: fmt.Errorf("OpenTelemetry protocol \"http/json\" is not supported"),
+		},
+	} {
+		t.Run(fmt.Sprintf("%v", testcase.input), func(t *testing.T) {
+			t.Logf("input: %v", testcase.input)
+
+			ctx := context.TODO()
+			_, err := newExporter(ctx, &testcase.input)
```

Comment:
should we also validate the returned "exporter" object here? instead of simply ignoring the returned value...

Reply:
Added to check exp when err is not nil.